cwms-data-api/src/main/java/cwms/cda/api/timeseriesprofile/TimeSeriesProfileParserCreateController.java

```diff
@@ -78,21 +79,25 @@ public void handle(@NotNull Context ctx) {
             boolean failIfExists = ctx.queryParamAsClass(FAIL_IF_EXISTS, Boolean.class).getOrDefault(true);
 
             TimeSeriesProfileParserDao tspParserDao = getParserDao(dsl);
+            String acceptHeader = ctx.header(Header.ACCEPT);
             String content = ctx.body();
             if (content.isEmpty()) {
                 ctx.status(HttpServletResponse.SC_BAD_REQUEST);
                 return;
             }
             if (content.contains(COLUMNAR_TYPE)) {
                 TimeSeriesProfileParserColumnar tspParserColumnar = Formats
-                        .parseContent(Formats.parseHeader(Formats.JSONV1, TimeSeriesProfileParserColumnar.class),
+                        .parseContent(Formats.parseHeader(acceptHeader, TimeSeriesProfileParserColumnar.class),
                                 content, TimeSeriesProfileParserColumnar.class);
                 tspParserDao.storeTimeSeriesProfileParser(tspParserColumnar, failIfExists);
-            } else {
+            } else if (content.contains(INDEXED_TYPE)) {
                 TimeSeriesProfileParserIndexed tspParserIndexed = Formats
-                        .parseContent(Formats.parseHeader(Formats.JSONV1, TimeSeriesProfileParserIndexed.class),
+                        .parseContent(Formats.parseHeader(acceptHeader, TimeSeriesProfileParserIndexed.class),
                                 content, TimeSeriesProfileParserIndexed.class);
                 tspParserDao.storeTimeSeriesProfileParser(tspParserIndexed, failIfExists);
+            } else {
+                ctx.status(HttpServletResponse.SC_BAD_REQUEST);
```

Comment:
Should add a message indicating the request body did not contain either columnar or indexed type ts profile parsers.

Reply:
Added error message with type error details

Reply:
you can just `throw new IllegalArgumentException()` with the message. It'll get transformed correctly with our exception handler.

Reply:
Replaced with exception throw with appropriate message